File: .claude/hooks/worktree-remove-cleanup.mjs
#!/usr/bin/env node
// @hookstack worktree-remove-cleanup
import { execSync } from "node:child_process";
// @hookstack worktree-remove-cleanup
// Nettoie un worktree supprimé : docker compose down + node_modules (WorktreeRemove)
import { existsSync, readFileSync, rmSync } from "node:fs";
import { join } from "node:path";
import { fileURLToPath } from "node:url";

function defaultExec(cmd) {
	execSync(cmd, { timeout: 30_000 });
}

export function run(
	input,
	{ exec = defaultExec, exists = existsSync, rm = rmSync } = {},
) {
	const p = input.worktree_path;
	if (!p) return null;

	const actions = [];

	try {
		if (
			exists(join(p, "docker-compose.yml")) ||
			exists(join(p, "docker-compose.yaml"))
		) {
			exec(
				`docker compose -f ${p}/docker-compose.yml down --remove-orphans 2>&1`,
			);
			actions.push("docker-down");
		}
	} catch {
		/* ignore */
	}

	try {
		const nm = join(p, "node_modules");
		if (exists(nm)) {
			rm(nm, { recursive: true, force: true });
			actions.push("rm-node-modules");
		}
	} catch {
		/* ignore */
	}

	return { actions };
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	const input = JSON.parse(readFileSync(0, "utf8"));
	run(input);
}

File: .claude/hooks/worktree-remove-cleanup.py
#!/usr/bin/env python3
# @hookstack worktree-remove-cleanup
"""Cleans up a removed worktree: docker compose down + node_modules (WorktreeRemove)."""
import json
import os
import shutil
import subprocess
import sys


def _default_exec(cmd):
    subprocess.run(cmd, shell=True, timeout=30, check=True)


def run(input_data=None, *, exec_cmd=None, exists=None, rm=None):
    if exec_cmd is None:
        exec_cmd = _default_exec
    if exists is None:
        exists = os.path.exists
    if rm is None:
        rm = lambda p, **kwargs: shutil.rmtree(p, ignore_errors=True)

    data = input_data or {}
    p = data.get("worktree_path")
    if not p:
        return None

    actions = []

    try:
        if exists(os.path.join(p, "docker-compose.yml")) or exists(
            os.path.join(p, "docker-compose.yaml")
        ):
            exec_cmd(f"docker compose -f {p}/docker-compose.yml down --remove-orphans 2>&1")
            actions.append("docker-down")
    except Exception:
        pass

    try:
        nm = os.path.join(p, "node_modules")
        if exists(nm):
            rm(nm, recursive=True, force=True)
            actions.append("rm-node-modules")
    except Exception:
        pass

    return {"actions": actions}


if __name__ == "__main__":
    data = json.load(sys.stdin)
    run(data)
